### PR TITLE
DAOS-5731 raft: Update to 0.7.1

### PR DESCRIPTION
**Pull request by liw**
Update raft to 0.7.1 that fixes recent Coverity issues.

PR-repos: raft@PR-40
Signed-off-by: Li Wei <wei.g.li@intel.com>

### CODE REVIEW

**Comment by liw** [comment]
Just FYI; I'll need to address the conflicts.

File: debian/control [hunk 1]
```diff
@@ -29,7 +29,7 @@ Build-Depends: debhelper (>= 10),
                libboost-dev,
                libspdk-dev,
                libipmctl-dev,
-               libraft-dev (= 0.7.0-1377.g4faff0b),
+               libraft-dev (= 0.7.1-1378.g264a878),
```
Comment (brianjmurrell) [inline debian/control:32]: It would be good to start tagging releases of raft so that we are not using git-hashed versions.

**Review by brianjmurrell — DISMISSED** [review]
Other than the need to rebase, to resolve conflicts, LGTM.

**Review by kccain — APPROVED** [review]

File: debian/control (hunk 1 above)
Comment (liw) [reply debian/control:32]: TBH, I'd rather us not making raft RPMs at all; just include raft submodule in daos source tarballs.

Comment (brianjmurrell) [reply debian/control]: The release process that we use in GitHub does not do this.  It does not have the ability to include submodules.

**Comment by brianjmurrell** [comment]
@liw 
> liw force-pushed the liw/coverity-raft branch from 8265727 to f4ad315 12 hours ago

Just a reminder that it's our group's policy to not  force-push once the review cycle has started as it invalidates peoples ability to do incremental reviews and requires reviewers to start from the beginning and fully review a patch once it has been force-pushed.

**Review by brianjmurrell — APPROVED** [review]

**Review by kccain — APPROVED** [review]

**Comment by liw** [comment]
> Just a reminder that it's our group's policy to not force-push once the review cycle has started as it invalidates peoples ability to do incremental reviews and requires reviewers to start from the beginning and fully review a patch once it has been force-pushed.

Given the amount and the conflict resolution nature of the change, I thought it'd be easier to just rebase this time. Sorry for the trouble.

**Comment by brianjmurrell** [comment]
> Given the amount and the conflict resolution nature of the change, I thought it'd be easier to review if I just rebased this time. Sorry for the trouble.

Fair enough.  It is a small change so not horrific.  Was just making sure that this was a carefully considered exception to the policy vs. just not knowing about it.

**Comment by liw** [comment]
rebuild 0-10 cont destroy and 18 version mismatch: DAOS-5851

**Pull request merged**